conn.go

```diff
@@ -552,7 +541,7 @@ func (c *Conn) handleAuth(arg string) {
 				c.WriteResponse(smtpErr.Code, smtpErr.EnhancedCode, smtpErr.Message)
 				return
 			}
-			c.WriteResponse(454, EnhancedCode{4, 7, 0}, err.Error())
+			c.WriteResponse(502, EnhancedCode{5, 7, 0}, err.Error())
```

Comment:
Hm, is this code more appropriate somehow?

Reply:
Right, on a second thought I think 454 is better. Reverted the change.

Reply:
Note, I haven't actually looked at the RFC, just wanted to know the motivation for the change.

Reply:
502 is completely unrelated to authentication whereas 454 is "temporary authentication failure". Not sure where I got 502 from.